src/celeritas/io/ImportOpticalMaterial.hh

```diff
@@ -34,17 +40,57 @@ struct ImportScintComponent
 
 //---------------------------------------------------------------------------//
 /*!
- * Store optical material properties for scintillation.
+ * Store material-only scintillation spectrum information.
+ * TODO: Components are not necessary in Geant4, but are in our generator.
  */
-struct ImportScintSpectrum
+struct ImportMaterialScintSpectrum
 {
-    double yield{};  //!< Characteristic light yield of the material
+    double yield{};  //!< Characteristic light yields of the material [1/MeV]
+    std::vector<ImportScintComponent> components;  //!< Fast/slow components
+
+    //! Whether all data are assigned and valid
+    explicit operator bool() const { return yield > 0 && !components.empty(); }
+};
+
+//---------------------------------------------------------------------------//
+/*!
+ * Store per-particle material scintillation spectrum information.
+ *
+ * The yield vector is the only necessary element, needed to calculate the
+ * yield based on the particle energy during the stepping loop.
+ * Components may not be assigned---they are the equivalent of
+ * \c k[Particle]ScintillationYield[i] in Geant4.
+ */
+struct ImportParticleScintSpectrum
+{
+    ImportPhysicsVector yield_vector;  //!< Particle yield vector
+    std::vector<ImportScintComponent> components;  //!< Fast/slow components
```

Comment:
Are particle-dependent `lambda_mean`, `lambda_sigma`, and `rise_time` used anywhere in scintillation? As far as I can tell only the `yield` and `fall_time` are needed.

Reply:
Not all the material-dependent components are in the Geant4 material properties table, but we do use them all in our scintillation generator. It seems like some of these particle-dependent components won't be used at all anywhere in the code.

Reply:
@whokion 's implementation is a bit different than the G4 one, so he could confirm if my thought process here is correct.

In the same way we have a fully defined scintillation spectrum for a given material, and we use all those values to sample our optical photons in the `ScintillationGenerator`, we'd have a fully defined spectrum for each particle type. Then, if we are sampling using a per-particle spectrum, the generator would loop over the components for that specific particle, and perform the same sampling, generating all the photons produced by that particle.

So yeah, the `ScintillationGenerator` will be updated in a future PR encompass this case. At that point, it will be able to sample the data based on a user-defined input, something like `use_material_spectrum{true/false};` as a part of our simulation input. And that will tell the generator if it should sample using material or particle data.

Reply:
Oh ok, so is this because we want to be able to sample the optical photon energies from different distributions for different particle types (different from Geant4 where the energy is sampled from the same distribution for all particles)?. Right now the difference between our implementations is that Geant4 samples from a tabulated distribution while we use the distribution parameters (`lambda_mean`, `lambda_sigma`) to sample from a Gaussian.

Reply:
> Now, Fast/slow are somewhat misleading

Fair. I'll update the comment.

> extended data with spectra[part][material_Id] compared to the current spectra[material] in ScintillationData

Yes, this is pretty much ready and it will be the next PR. Just waiting for this one to be merged in.

> However, I am really doubt whether there are noticeable differences in scintillation yields or models per particle type for the given same amount of energy loss over a step

This is a point that I thought too, but I settled with: if Geant4 had different yields for a handful of particles, they must be non-negligible and useful to someone, otherwise it would have been a moot point implementing them. And I would guess a scintillation caused by a neutron via strong force is different from a scintillation caused by a muon, which is a MIP.

Reply:
@stognini The question is whether one (who) really need to treat scintillation differently for different particles (major ones are likely electron vs. hadron) at the same application - i.e., the yield is not just a function of the energy loss, but also the particle type (for example, the energy loss by protons is treated differently from the same amount of energy loss by electrons). The neutron (neutral) should not be a part of the picture and photons produced by the deexcitation by the neutron-nucleus should be considered regular photons, not optical photons. Regarding to MIP, the real question is whether number of optical photons produced by the same amount of energy loss (per unit length) between muons and hadrons are different (does not matter whether MIP by muons or hadrons)?  Maybe yes, but how much, and who cares about their differences.  Why does one need such a fine detail simulation for what application?  As far as I know, Geant4 introduced this particle dependency recently, but the motivation is not clear, especially for HEP (in which the timing is more important than the number of photons collected in a photo-detector, i.e, intensity which can be always amplified by a photo-multiplier).